main.go

```diff
@@ -123,6 +124,9 @@ func InitFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&clusterTopologyConcurrency, "clustertopology-concurrency", 10,
 		"Number of clusters to process simultaneously")
 
+	fs.IntVar(&machineDeploymentTopologyConcurrency, "machinedeploymenttopology-concurrency", 10,
+		"Number of machine deployments to process simultaneously")
```

Comment:
Do we need to expose this option? We're not really reconciling anything other than cleanup of resources.

In alternative to this approach, given that we have a feature gate, should we consider adding extra logic to MD and MS reconcilers instead?

Reply:
Just added the option because we were doing it for all other reconcilers. I don't think it's necessary.

If it's okay to the logic to the other controllers I would prefer it. I just thought we want to explicitly keep topology reconciliation separate.

Reply:
I'm in favour of keeping everything topology related separated from existing controllers, but if this is considered overkilling for template deletion, I'm ok in making an exception in this case

Reply:
Benefit would also be that we only have 1 reconciler for MD/MS each. 
Although I think it wouldn't be a big problem if we have multiple. They mostly shouldn't interfere as:
* the existing reconciler do nothing after the deletionTimeStamp is set
* most of the new reconciler reconciles only when the deletionTimeStamp is set. Though there is a small overlap when the finalizer is set. So I assume when a MachineSet is created both would be triggered and the finalizer addition would be faster. Not sure if that leads to problematic conflicts when the regular machineset reconciler tries to update the MachineSet.

Reply:
Dropped the flag. Also separated the reconcilers, but kept them separate (for now). Will push later.